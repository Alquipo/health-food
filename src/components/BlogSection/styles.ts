import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.section`
  ${({ theme }) => css`
    background: ${theme.colors.lightGray};
    overflow: hidden;
  `}
`

export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: ${theme.spacings.xxlarge} 0 ${theme.spacings.large};
  `}
`

export const Header = styled.div`
  width: 100%;
  max-width: 51rem;
`

export const Title = styled.h2`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xxlarge};
    color: ${theme.colors.secondary};
    text-align: center;
  `}
`

export const Text = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.large};
    margin-top: ${theme.spacings.small};
    color: ${theme.colors.text};
    text-align: center;
    line-height: 1.4;
  `}
`

export const SectionGallery = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.grid.container};

    ${media.greaterThan('medium')`
      margin-bottom: calc(${theme.spacings.xlarge} * 2);

      display: block;

    `};
  `}
`
