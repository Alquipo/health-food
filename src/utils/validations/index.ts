import * as yup from 'yup'

const nascimentoRegex =
  /^((((19|[2-9]\d)\d{2})-(0[13578]|1[02])-(0[1-9]|[12]\d|3[01]))|(((19|[2-9]\d)\d{2})-(0[13456789]|1[012])-(0[1-9]|[12]\d|30))|(((19|[2-9]\d)\d{2})-02-(0[1-9]|1\d|2[0-8]))|(((1[6-9]|[2-9]\d)(0[48]|[2468][048]|[13579][26])|((16|[2468][048]|[3579][26])00))-02-29))$/g

const cpfRegex = /^\d{3}.?\d{3}.?\d{3}-?\d{2}$/

const cepRegex = /^([\d]{2})\.*([\d]{3})-*([\d]{3})/

const estadoRegex =
  /^(AC|AL|AP|AM|BA|CE|DF|ES|GO|MA|MT|MS|MG|PA|PB|PR|PE|PI|RJ|RN|RS|RO|RR|SC|SP|SE|TO)$/i

export const fieldsValidationSchema = yup.object({
  name: yup.string().required('Is not allowed to be empty'),
  birthDate: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(nascimentoRegex, 'must be a valid email'),
  cpf: yup
    .string()
    .max(14, 'Must be 14 characters')
    .required('Is not allowed to be empty')
    .matches(cpfRegex, 'Invalid CPF'),
  zipCode: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(cepRegex, 'Invalid ZipCode'),
  address: yup.string().required('Is not allowed to be empty'),
  district: yup.string().required('Is not allowed to be empty'),
  number: yup.number().required('Is not allowed to be empty'),
  city: yup.string().required('Is not allowed to be empty'),
  state: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(estadoRegex, 'Invalid State')
})
