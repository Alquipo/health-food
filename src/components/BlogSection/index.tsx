import * as S from './styles'

import Gallery from 'components/Gallery'
import gallery from '../Gallery/mock'

export default function BlogSection() {
  return (
    <S.Wrapper>
      <S.Content>
        <S.Header>
          <S.Title>Read Our Blog</S.Title>

          <S.Text>
            Far far away, behind the word mountains, far from the countries
            Vokalia and Consonantia, there live the blind texts.
          </S.Text>
        </S.Header>

        <S.SectionGallery>
          <Gallery items={gallery} />
        </S.SectionGallery>
      </S.Content>
    </S.Wrapper>
  )
}
