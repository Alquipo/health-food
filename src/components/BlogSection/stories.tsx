import { Story, Meta } from '@storybook/react'
import BlogSection from '.'

export default {
  title: 'BlogSection',
  component: BlogSection
} as Meta

export const Default: Story = () => <BlogSection />
