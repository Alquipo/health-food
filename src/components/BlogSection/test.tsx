import { render, screen } from 'utils/test-utils'

import BlogSection from '.'

jest.mock('components/Gallery', () => ({
  __esModule: true,
  default: function Mock() {
    return <div data-testid="Mock Gallery" />
  }
}))

describe('<BlogSection/>', () => {
  it('should render a Blog Section correctly', () => {
    const { container } = render(<BlogSection />)

    expect(
      screen.getByRole('heading', { name: /Read Our Blog/i })
    ).toBeInTheDocument()

    expect(
      screen.getByText(
        /Far far away, behind the word mountains, far from the countries Vokalia and Consonantia, there live the blind texts./i
      )
    ).toBeInTheDocument()

    expect(screen.getAllByTestId(/mock Gallery/i)).toHaveLength(1)

    expect(container.firstChild).toMatchSnapshot()
  })
})
