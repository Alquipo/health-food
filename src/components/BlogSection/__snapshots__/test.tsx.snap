// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogSection/> should render a Blog Section correctly 1`] = `
.c0 {
  background: #FAFAFC;
  overflow: hidden;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 5.6rem 0 4.0rem;
}

.c2 {
  width: 100%;
  max-width: 51rem;
}

.c3 {
  font-size: 3.6rem;
  color: #1D164D;
  text-align: center;
}

.c4 {
  font-size: 2.0rem;
  font-weight: 600;
  margin-top: 2.4rem;
  color: #9E9BAF;
  text-align: center;
}

.c5 {
  width: 100%;
  max-width: 160rem;
}

@media (max-width:450px) {
  .c3 {
    font-size: 2.8rem;
  }
}

@media (max-width:450px) {
  .c4 {
    font-size: 1.8rem;
  }
}

@media (min-width:768px) {
  .c5 {
    margin-bottom: calc(4.8rem * 2);
    display: block;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h2
        class="c3"
      >
        Read Our Blog
      </h2>
      <p
        class="c4"
      >
        Far far away, behind the word mountains, far from the countries Vokalia and Consonantia, there live the blind texts.
      </p>
    </div>
    <div
      class="c5"
    >
      <div
        data-testid="Mock Gallery"
      />
    </div>
  </div>
</section>
`;
